### PR TITLE
Update strategies doc

### PR DESCRIPTION
**Pull request by jssblck**
# Overview

It was brought to our attention that the docs for nuget strategies were incorrect: they said the nuget strategies are dynamic but they're actually static.

This PR updates that listing and overhauls the table in an attempt to make the dynamic vs static relationship more clear. Note: I did not double check every strategy; I just went off what we currently listed in that doc.

[Rendered preview here.](https://github.com/fossas/fossa-cli/tree/docs/nuget/docs/references/strategies#strategies-by-type)

Screenshot 2024-11-27 at 1 02 00 PM

### CODE REVIEW

File: docs/references/strategies/README.md [hunk 1]
```diff
@@ -145,40 +145,49 @@ Invoke strict analysis with the `--strict` flag when running `fossa analyze`.
...
+> [!TIP]
+> If FOSSA CLI is forced to utilize a fallback strategy, meaning it did not detect ideal results,
+> a warning is emitted in the scan summary after running `fossa analyze`.
+
+> [!WARNING]
+> "Custom" strategies work very differently than the standard package manager based analysis; read their docs for more details.
+
+| Language/Package Manager                                                                                                                        | Kind of analysis             | Detect Vendored Code |
+|-------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------|----------------------|
+| [C#](https://github.com/fossas/fossa-cli/tree/master/docs/references/strategies/languages/dotnet)                                               | Static                       | ❌                    |
```
Comment (csasarak) [inline docs/references/strategies/README.md:164]: [Question] Is Nuget always what's used for C#? I don't need this resolved for approval but it feels like the package manager or NuGet is the right thing to put here vs. the language.

**Review by csasarak — APPROVED** [review]

File: docs/references/strategies/README.md (hunk 1 above)
Comment (jssblck) [reply docs/references/strategies/README.md:164]: it's a little strange. first: no, nuget isn't the only package manager for C# or .NET generally. But inside FOSSA CLI we group all the tactics for .NET under the `Nuget` module, e.g. `Strategy.NuGet.Paket`.

i do think it probably makes more sense to refer to the package manager in this table like we do for other languages.

Comment (jssblck) [reply docs/references/strategies/README.md]: I've gone ahead and updated these to match the other package managers in the table: C# was the only odd one out.

Comment (csasarak) [reply docs/references/strategies/README.md]: I think that looks good, thanks!

**Pull request merged**